packages/sdk/app-framework/src/plugins/next/manager.ts

```diff
@@ -0,0 +1,305 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+import { Event } from '@dxos/async';
+import { create, type ReactiveObject } from '@dxos/live-object';
+import { log } from '@dxos/log';
+import { type MaybePromise } from '@dxos/util';
+
+import { type ActivationEvent, type AnyCapability, type Plugin, type PluginModule, PluginsContext } from './plugin';
+
+export const eventKey = (event: ActivationEvent) => (event.specifier ? `${event.id}:${event.specifier}` : event.id);
+
+export type PluginManagerOptions = {
+  pluginLoader: (id: string) => MaybePromise<Plugin>;
+  plugins?: Plugin[];
+  core?: string[];
+};
+
+export class PluginManager {
+  readonly activation = new Event<{ event: string; state: 'activating' | 'activated' | 'error'; error?: any }>();
+
+  private readonly _context = new PluginsContext({
+    activate: (event) => this.activate(event),
+    reset: (id) => this.reset(id),
+  });
+
+  private readonly _state: ReactiveObject<{
+    core: string[];
+    enabled: string[];
+    active: string[];
+    eventsFired: string[];
+    pendingReset: string[];
+  }>;
+
+  private readonly _pluginLoader: PluginManagerOptions['pluginLoader'];
+  private readonly _plugins = new Map<string, Plugin>();
+  private readonly _modules = new Map<string, PluginModule>();
+  private readonly _capabilities = new Map<string, AnyCapability[]>();
+
+  constructor({ pluginLoader, plugins = [], core = [] }: PluginManagerOptions) {
+    this._pluginLoader = pluginLoader;
+    this._state = create({
+      core,
+      enabled: plugins.map((plugin) => plugin.meta.id),
+      active: [],
+      eventsFired: [],
+      pendingReset: [],
+    });
+    plugins.forEach((plugin) => this._registerPlugin(plugin));
+  }
+
+  get context() {
+    return this._context;
+  }
+
+  /**
+   * Ids of plugins that are core and cannot be removed.
+   */
+  get core(): readonly string[] {
+    return this._state.core;
+  }
+
+  /**
+   * Ids of plugins that are currently registered.
+   */
+  get enabled(): readonly string[] {
+    return this._state.enabled;
+  }
+
+  /**
+   * Ids of modules that are currently active.
+   */
+  get active(): readonly string[] {
```

Comment:
are those arrays reactive?

how do you feel about wrapping them like `ReactiveObject<readonly string[]>` (to be renamed to just Live) so its explicit in the api?

Reply:
they are and we probably should do that